smartcontactmanager/src/main/java/com/scm/smartcontactmanager/service/category/CategoryServiceImpl.java

```diff
@@ -1,8 +1,137 @@
+		CategoryResponseDto categoryResponseDto = new CategoryResponseDto();
+		categoryResponseDto.setName(categoryResponseDto.getName());
+		categoryResponseDto.setType(categoryResponseDto.getType());
+		categoryResponseDto.setUserId(categoryResponseDto.getUserId());
+		categoryResponseDto.setByAdmin(false);
+		return categoryResponseDto;
+	}
+	
+	
+	@Override
+	public List<Category> getListOfCategory(){
+		List<Category> categoryList = categoryDao.getListOfCategory();
+		if(categoryList==null) {
+			throw new ServiceException(HttpStatusCode.valueOf(404), "Categories not found");
+		}
+		return categoryList;
+	}
 
+	@Override
+	public void addContactsToCategory(int id, ContactIds contactIds) throws Exception {
+		Category category = categoryDao.getCategoryById(id).get();
+		List<Contact> contactList = categoryDao.getContactdByIds(contactIds.getContactId());
+		if(category != null && contactList.size() == contactIds.getContactId().size()) {
+			List<ContactCategoryRelation> relations = new ArrayList<>();
+			for(int cId: contactIds.getContactId()) {
+				ContactCategoryRelation relation = ContactCategoryRelation.builder()
+						                                       .categoryId(id)
+						                                       .contactId(cId)
+						                                       .build();
+		
+				relations.add(relation);
+			}
+			List<ContactCategoryRelation> concatrelation = categoryDao.addContactsToCategory(relations);
+		} else {
+			throw new Exception("Invalid Input!");
+		}
+		
+	}
+    @Override
+	public  List<Contact> getContactsByCategoryId(int id) {
+    	
+    	Optional<Category> categoryById = categoryDao.getCategoryById(id);	
+    	if(categoryById.isEmpty()) {
+    		throw new ServiceException(HttpStatusCode.valueOf(404),"Category not found with id: "+id);
+    	}else {
+    	List<ContactCategoryRelation> relations = categoryDao.getContactCategoryRelation(id);
+    	List<Integer> contactIds = relations.stream().map(rl -> rl.getContactId()).collect(Collectors.toList());
+//    	List<Integer> contactIds = new ArrayList<>();
+//    	for(ContactCategoryRelation rl: relations) {
+//    		contactIds.add(rl.getContactId());
+//    	}
+    	List<Contact> contactList = categoryDao.getContactdByIds(contactIds);
+    	return contactList;
+    	}
+		
+	}
+	@Override
+	public void deleteCategoryById(int id) {
+		Optional<Category> categoryById = categoryDao.getCategoryById(id);	
+    	if(categoryById.isEmpty()) {
+    		throw new ServiceException(HttpStatusCode.valueOf(404),"Category not found with id: "+id);
+    	}
+//		List<ContactCategoryRelation> relations = categoryDao.getContactCategoryRelation(id);
+//		List<Integer> contactIds = relations.stream().map(rd ->rd.getContactId()).collect(Collectors.toList());
+//      categoryDao.deleteContactByIds(contactIds);
+    	categoryDao.deleteCategoryById(id);
+    	List<ContactCategoryRelation> findByCategoryId = categoryDao.getContactCategoryRelation(id);
```

Comment:
Variable name should be existingContactCategoryRelations

Reply:
Addressed